beekeeper-integration-tests/src/test/java/com/expediagroup/beekeeper/integration/api/BeekeeperApiIntegrationTest.java

```diff
@@ -181,11 +181,107 @@ public void testGetPathsWhenThereIsFiltering() throws SQLException, InterruptedE
     Page<HousekeepingPathResponse> responsePage = mapper
         .readValue(body, new TypeReference<RestResponsePage<HousekeepingPathResponse>>() {});
     List<HousekeepingPathResponse> result = responsePage.getContent();
-
+    assertThat(responsePage.getTotalElements()).isEqualTo(1L);
     assertThatPathsEqualsResponse(testPath1, result.get(0));
     assertThat(result.size()).isEqualTo(1);
   }
 
+  @Test
+  public void testPathsPageable() throws SQLException, InterruptedException, IOException {
+    // Test the total elements on different pages by setting the page size to 1.
+    int pageSize = 1;
+
+    // Create three new HousekeepingPath objects
+    HousekeepingPath testPath1 = createHousekeepingPath("some_table",
+        "s3://some/path/event_date=2020-01-01/event_hour=0/event_type=A", LifecycleEventType.EXPIRED,
+        Duration.parse("P3D").toString());
+    HousekeepingPath testPath2 = createHousekeepingPath("some_table",
+        "s3://some/path/event_date=2020-01-01/event_hour=0/event_type=B", LifecycleEventType.UNREFERENCED,
+        Duration.parse("P3D").toString());
+    HousekeepingPath testPath3 = createHousekeepingPath("some_table",
+        "s3://some/path/event_date=2020-01-01/event_hour=0/event_type=D", LifecycleEventType.UNREFERENCED,
+        Duration.parse("P3D").toString());
+
+    // Set the housekeepingStatus and cleanupTimestamp properties
+    testPath1.setHousekeepingStatus(HousekeepingStatus.FAILED);
+    testPath2.setHousekeepingStatus(HousekeepingStatus.FAILED);
+    testPath3.setHousekeepingStatus(HousekeepingStatus.FAILED);
+    testPath1.setCleanupTimestamp(LocalDateTime.parse("1999-05-05T10:41:20"));
+    testPath2.setCleanupTimestamp(LocalDateTime.parse("1999-05-05T10:41:20"));
+    testPath3.setCleanupTimestamp(LocalDateTime.parse("1999-05-05T10:41:20"));
+
+    // Insert the three objects into the database
+    insertUnreferencedPath(testPath1);
+    insertUnreferencedPath(testPath2);
+    insertUnreferencedPath(testPath3);
+
+    // Call the getUnreferencedPaths() method with the relevant filters
+    String filters = "?housekeeping_status=FAILED&page=1&size=" + pageSize; // Concat the pageSize set initially
+    HttpResponse<String> response = testClient.getUnreferencedPaths("some_database", "some_table", filters);
+
+    // Assert that the response contains the entries in the database
+    assertThat(response.statusCode()).isEqualTo(OK.value());
+    String body = response.body();
+    Page<HousekeepingPathResponse> responsePage = mapper
+        .readValue(body, new TypeReference<RestResponsePage<HousekeepingPathResponse>>() {});
+    List<HousekeepingPathResponse> result = responsePage.getContent();
+
+    // Assert that each page contains exactly one element
+    assertThat(result.size()).isEqualTo(1);
```

Comment:
you're checking only the first page here, you could iterate through all 2 pages and check this for  all of them